stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/images/AttachmentGallery.kt

```diff
@@ -37,13 +57,22 @@ public class AttachmentGallery : ConstraintLayout {
     }
 
     public fun provideImageList(fragmentActivity: FragmentActivity, imageList: List<String>, currentIndex: Int = 0) {
-        binding.attachmentGallery.adapter = AttachmentSlidePagerAdapter(fragmentActivity, imageList)
+        adapter = AttachmentSlidePagerAdapter(fragmentActivity, imageList)
+        binding.attachmentGallery.adapter = adapter
         configPositionCount(imageList.size)
         binding.attachmentGallery.setCurrentItem(currentIndex, false)
+        binding.shareButton.setOnClickListener {
+            CoroutineScope(DispatcherProvider.IO).launch {
```

Comment:
This can be `GlobalScope.launch(DispatcherProvider.Main) { ... }` and then we can drop the `withContext` call later on. `ImageLoader` has proper suspending functions, so it won't block the main thread.